lib/Driver/Driver.cpp

```diff
@@ -1803,14 +1764,10 @@ static StringRef getOutputFilename(Compilation &C,
 
   auto ShouldPreserveOnSignal = PreserveOnSignal::No;
 
-  // FIXME: Treat GeneratePCHJobAction as non-top-level (to get tempfile and not
-  // use the -o arg) even though, based on ownership considerations within the
-  // driver, it is stored as a "top level" JobAction.
   if (auto *PCHAct = dyn_cast<GeneratePCHJobAction>(JA)) {
     // For a persistent PCH we don't use an output, the frontend determines
     // the filename to use for the PCH.
     assert(!PCHAct->isPersistentPCH());
-    AtTopLevel = false;
```

Comment:
Hm, at one point I tried to remove the distinction between top-level actions and non-top-level actions and failed. I'll look into this again before merging.